### ISSUE TITLE
error: could not build Objective-C module 'My_Framework'

### ISSUE THREAD

**Issue by VetriThiyagu** [CocoaPods/CocoaPods#10025]
* [ ] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

I am creating a Cocoapod framework which includes c++ static library, objective-C and Swift classes, none of these are 3rd party libraries, I needed to include one more 3rd party static framework "AdManager" which is compiled using Carthage, I added "AdManager/FWSDK.h" in my modulemap file , It was throwing an error "error: could not build Objective-C module 'My_Framework'". So could you please help me to get through?

## What did you expect to happen?

Build Successful

## What happened instead?

<module-includes>:1:9: note: in file included from <module-includes>:1:
    #import "Headers/MMSmartStreamingWithFreewheel-umbrella.h"
            ^
    //private/var/folders/cq/ybgs7brs1cl0h9dv4q0mwpdm0000gn/T/CocoaPods-Lint-20200902-14958-q7q7x-MMSmartStreamingWithFreewheel/Pods/Target Support Files/MMSmartStreamingWithFreewheel/MMSmartStreamingWithFreewheel-umbrella.h:14:9: note: in file included from //private/var/folders/cq/ybgs7brs1cl0h9dv4q0mwpdm0000gn/T/CocoaPods-Lint-20200902-14958-q7q7x-MMSmartStreamingWithFreewheel/Pods/Target Support Files/MMSmartStreamingWithFreewheel/MMSmartStreamingWithFreewheel-umbrella.h:14:
    #import "MMSmartStreamingHeader.h"
            ^
    /Users/macair1/Desktop/CocoaPod/MMSmartStreamingWithFreewheel/MMSmartStreamingWithFreewheel/Classes/Common/MMSmartStreaming/MMSmartStreamingHeader.h:11:9: error: include of non-modular header inside framework module 'MMSmartStreamingWithFreewheel.MMSmartStreamingHeader': '/Users/macair1/Desktop/CocoaPod/MMSmartStreamingWithFreewheel/MMSmartStreamingWithFreewheel/Classes/Frameworks/AdManager.framework/Headers/FWSDK.h'
    #import "AdManager/FWSDK.h"
            ^
    <module-includes>:1:9: note: in file included from <module-includes>:1:
    #import "Headers/MMSmartStreamingWithFreewheel-umbrella.h"
            ^
    //private/var/folders/cq/ybgs7brs1cl0h9dv4q0mwpdm0000gn/T/CocoaPods-Lint-20200902-14958-q7q7x-MMSmartStreamingWithFreewheel/Pods/Target Support Files/MMSmartStreamingWithFreewheel/MMSmartStreamingWithFreewheel-umbrella.h:17:9: error: include of non-modular header inside framework module 'MMSmartStreamingWithFreewheel': '/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator13.2.sdk/usr/lib/swift/shims/module.modulemap'
    #import "module.modulemap"
            ^
    <module-includes>:1:9: note: in file included from <module-includes>:1:
    #import "Headers/MMSmartStreamingWithFreewheel-umbrella.h"
            ^
    //private/var/folders/cq/ybgs7brs1cl0h9dv4q0mwpdm0000gn/T/CocoaPods-Lint-20200902-14958-q7q7x-MMSmartStreamingWithFreewheel/Pods/Target Support Files/MMSmartStreamingWithFreewheel/MMSmartStreamingWithFreewheel-umbrella.h:17:9: note: in file included from //private/var/folders/cq/ybgs7brs1cl0h9dv4q0mwpdm0000gn/T/CocoaPods-Lint-20200902-14958-q7q7x-MMSmartStreamingWithFreewheel/Pods/Target Support Files/MMSmartStreamingWithFreewheel/MMSmartStreamingWithFreewheel-umbrella.h:17:
    #import "module.modulemap"
            ^
    /Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator13.2.sdk/usr/lib/swift/shims/module.modulemap:1:1: error: unknown type name 'module'
    module SwiftShims {
    ^
    <module-includes>:1:9: note: in file included from <module-includes>:1:
    #import "Headers/MMSmartStreamingWithFreewheel-umbrella.h"
            ^
    //private/var/folders/cq/ybgs7brs1cl0h9dv4q0mwpdm0000gn/T/CocoaPods-Lint-20200902-14958-q7q7x-MMSmartStreamingWithFreewheel/Pods/Target Support Files/MMSmartStreamingWithFreewheel/MMSmartStreamingWithFreewheel-umbrella.h:17:9: note: in file included from //private/var/folders/cq/ybgs7brs1cl0h9dv4q0mwpdm0000gn/T/CocoaPods-Lint-20200902-14958-q7q7x-MMSmartStreamingWithFreewheel/Pods/Target Support Files/MMSmartStreamingWithFreewheel/MMSmartStreamingWithFreewheel-umbrella.h:17:
    #import "module.modulemap"
            ^
    /Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator13.2.sdk/usr/lib/swift/shims/module.modulemap:1:18: error: expected ';' after top level declarator
    module SwiftShims {
                     ^
    <unknown>:0: error: could not build Objective-C module 'MMSmartStreamingWithFreewheel'
    
    warning: Skipping code signing because the target does not have an Info.plist file and one is not being generated automatically. (in target 'App' from project 'App')
    ** BUILD FAILED **

## CocoaPods Environment

Pod::Spec.new do |s|
  
  s.swift_version    = '4.0'
  s.ios.deployment_target = '8.0'
  s.tvos.deployment_target = '9.0'
  s.source_files = 'MMSmartStreamingWithFreewheel/Classes/Common/**/*.{h,swift}'
  s.ios.source_files = 'MMSmartStreamingWithFreewheel/Classes/iOS/*.{h,swift}'
  s.tvos.source_files = 'MMSmartStreamingWithFreewheel/Classes/tvOS/*.{h,swift}'
  s.frameworks = 'UIKit', 'AVFoundation', 'AdSupport', 'CoreLocation', 'CoreGraphics', 'SystemConfiguration'
  s.ios.frameworks = 'CoreTelephony', 'Messages', 'MessagesUI', 'WebKit', 'EvenKit'
  s.ios.vendored_libraries = 'MMSmartStreamingWithFreewheel/Classes/iOS/libmmsmartstreamer.a'
  s.tvos.vendored_libraries = 'MMSmartStreamingWithFreewheel/Classes/tvOS/libmmsmartstreaming-tvos.a'
  s.libraries = 'stdc++', 'xml2', 'xml2.2'
  s.public_header_files = 'MMSmartStreamingWithFreewheel/Classes/Common/MMSmartStreaming/**/*.h' , 'MMSmartStreamingWithFreewheel/Classes/Common/ReachabilityMM/*.h', 'MMSmartStreamingWithFreewheel/Classes/Frameworks/AdManager.framework/Headers/*.h', 'MMSmartStreamingWithFreewheel/Classes/Common/MMSmartStreaming/module.modulemap'
  s.static_framework = true
  s.vendored_frameworks = 'MMSmartStreamingWithFreewheel/Classes/Frameworks/AdManager.framework'
  s.pod_target_xcconfig = { 'OTHER_LDFLAGS' => ['-lObjC'] }
end

## Project that demonstrates the issue

As I'm new to the CocoaPods, Please help me to get through, I have been stuck in this issue for long time

**Comment by dnkoutso** [comment]
I am sorry but I think it is best to ask in StackOverflow for assistance here. It is hard to parse whether this is a CocoaPods library issue to be honest.

Please upload a sample app, and very clear reproducible steps for us to be able to diagnose.

**Comment by VetriThiyagu** [comment]
Hi @dnkoutso ,
Thank you verymuch for the reply, As you are asking, am uploading my sample app here, 

https://bitbucket.org/vetri-mediamelon/mmsmartstreamingwithfreewheel.git

to reproduce, I did only 

pod lib lint MMSmartStreamingWithFreewheel.podspec --verbose

**Comment by VetriThiyagu** [comment]
Hi @dnkoutso ,
Did you find the issue?

**Comment by VetriThiyagu** [comment]
Hi @dnkoutso ,
Any luck with the issue?

**Comment by dnkoutso** [comment]
@VetriThiyagu you must remove:
```
'MMSmartStreamingWithFreewheel/Classes/Common/MMSmartStreaming/module.modulemap'
```

From `s.public_header_files` and instead use `s.module_map` if you want to provide your own custom module map.

Then you need to fix `MMSmartStreamingPrivate` inside your `module.modulemap` to find the correct header. It should probably be `umbrella header "FWSDK.h"` or something like that.

I am going to close this ticket as so far I do not see any issue with CocoaPods library. I _will_ reopen it though if you are stuck and believe you have found a bug in CocoaPods.
